### PR TITLE
feat: Allow setting of EncMode and ECLevel via ImageOption

### PR DESCRIPTION
**Pull request by fionera**
Currently its not possible to specify the encoding mode. This change adds the option for this and also allows setting the error correction level without specifying the version

### CODE REVIEW

File: image_option_api.go [hunk 1]
```diff
@@ -163,3 +163,18 @@ func WithCustomImageEncoder(encoder ImageEncoder) ImageOption {
 		oo.imageEncoder = encoder
 	})
 }
+
+// WithEncodingMode to specify which encMode to use
+func WithEncodingMode(mode encMode) ImageOption {
```
Comment (yeqown) [inline image_option_api.go:168]: This is not an option for the output image, consider other input param plz ~

File: image_option_api.go [hunk 2]
```diff
@@ -163,3 +163,18 @@ func WithCustomImageEncoder(encoder ImageEncoder) ImageOption {
...
+	return newFuncDialOption(func(oo *outputImageOptions) {
+		oo.encMode = mode
+	})
+}
+
+// WithBuiltinImageEncoder to use custom image encoder to encode image.Image into
+// io.Writer
+func WithErrorCorrectionLevel(level ecLevel) ImageOption {
```
Comment (yeqown) [inline image_option_api.go:176]: same as the previous (encMode)

**Review by yeqown — COMMENTED** [review]
I got your meaning, but it looks weird to put `enMode` and `ecLevel` into `ImageOption`, consider about add a `NewXXX` API instead plz.

**Comment by orefalo** [comment]
@yeqown if I understand you comments rights,
you are recommending to create a new options data structure different from the one that controls the image generation.

Call it QrGenerationOptions for instance, and in it we can define there attributes..

**Comment by yeqown** [comment]
@orefalo that's it, `QrGenerationOptions` only used in new API or old API would be broken. 

In my thought, just add one API `NewWithConfig(src string, cfg *GenerateConfig, opts ...OutputImageOption)` is the best way  to achieve this.

**Comment by orefalo** [comment]
so I started the implementations with two different data structure.
IMHO it complexities the API - I am not sure if it's a good idea. Would be so much easier to have one config object with all attributes.

Last, I need help ( as I am no golang guru)
```
func NewWithSpecV(text string, ver int, ecLv ecLevel, imgOpts ...ImageOption, encOpts ...EncoderOption) 
```

the above code fails b/c ... can only be used as the last arg. What would be your recommendation @yeqown?

To me this is a clear demonstration that a single structure is easier to manage.

**Comment by yeqown** [comment]
one more `Variable parameter(imgOpts and encOpts)` is not allowed in one function in `Go`, so this is invalid:

```go
func NewWithSpecV(text string, ver int, ecLv ecLevel, imgOpts ...ImageOption, encOpts ...EncoderOption) 
```

Think about this:

```go
cfg := qrcode.NewConfig() // returns default value
cfg.ApplyEcLevel(qr.EvLevel_HIGH)
cfg.ApplyEncMode(qrcpode.EncModeNumeric)
// more config options...

NewWithConfig(src string, cfg *GenerateConfig, opts ...OutputImageOption)
```

**Comment by orefalo** [comment]
Hi, sorry for the delay, I got distracted. I created pull req #32 based on your recommendations. please review. Sincerely

**Pull request closed without merging**